detect/detect.go

```diff
@@ -398,8 +398,10 @@ MatchLoop:
 			} else {
 				// If |secretGroup| is not set, we will use the first suitable capture group.
 				if len(groups) == 2 {
-					// Use the only group.
-					finding.Secret = groups[1]
+					if s := groups[1]; s != "" {
+						// Use the only group.
+						finding.Secret = s
+					}
 				} else {
 					// Use the first non-empty group.
 					for _, s := range groups[1:] {
```

Comment:
This is a good change and should fix the empty secret, but unless I am mistaken, now both branches of the `if len(groups)==2` do the same thing?

So this should be equivalent
```patch
diff --git a/detect/detect.go b/detect/detect.go
index ac72498..8e9131e 100644
--- a/detect/detect.go
+++ b/detect/detect.go
@@ -396,22 +396,15 @@ MatchLoop:
 				}
 				finding.Secret = groups[rule.SecretGroup]
 			} else {
 				// If |secretGroup| is not set, we will use the first suitable capture group.
-				if len(groups) == 2 {
-					if s := groups[1]; s != "" {
-						// Use the only group.
-						finding.Secret = s
-					}
-				} else {
 					// Use the first non-empty group.
 					for _, s := range groups[1:] {
 						if len(s) > 0 {
 							finding.Secret = s
 							break
 						}
 					}
-				}
 			}
 		}
 
 		// check if the regexTarget is defined in the allowlist "regexes" entry

```

Reply:
Hah, good point.